gocourse10/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/senchov/go_hausaufgaben/gocourse10/observer"
+)
+
+func main() {
+	fmt.Println("Start")
+	// strategy
+	collar := Collar{Trough: MeetTrought{}, ID: 1}
+	Feed(collar)
+	collar.SetTrough(&FishTrough{})
+	Feed(collar)
+	collar.SetTrough(&VegetablesTrought{})
+	Feed(collar)
+
+	//decorator
```

Comment:
["Comments for humans always have a single space after the slashes"](https://dmitri.shuralyov.com/idiomatic-go#comments-for-humans-always-have-a-single-space-after-the-slashes)

```suggestion
	// decorator
```